packages/provider-utils/src/media-type-to-extension.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Maps a media type to its corresponding file extension.
+ * It was originally introduced to set a filename for audio file uploads
+ * in https://github.com/vercel/ai/pull/8159.
+ *
+ * @param mediaType The media type to map.
+ * @returns The corresponding file extension
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/MIME_types/Common_types
+ */
+export function mediaTypeToExtension(mediaType: string) {
+  const [_type, subtype] = mediaType.split('/');
+
+  return (
+    {
+      mpeg: 'mp3',
+      'x-wav': 'wav',
+      opus: 'ogg',
+      mp4: 'm4a',
+      'x-m4a': 'm4a',
+    }[subtype] || subtype
+  );
+}
```

Comment:
Should we somehow combine this with https://github.com/vercel/ai/blob/7729e322da009d0fd34ea14a768dbc8f0e2b5520/packages/ai/src/util/detect-media-type.ts?

Reply:
not sure; it seems to be separate unless i am missing something.

if we have the bytes i prefer detecting the type from the bytes, it is much more robust (this should happen on the `ai` level i.e. `transcribe` already)